src/pip/_internal/compat.py

```diff
@@ -192,3 +193,43 @@ def samefile(file1, file2):
         path1 = os.path.normcase(os.path.abspath(file1))
         path2 = os.path.normcase(os.path.abspath(file2))
         return path1 == path2
+
+
+if sys.version_info >= (3, 3):
```

Comment:
Rather than an explicit version check, would it be better to do `if hasattr(shutil, 'get_terminal_size')`? Checking directly for what we need will be more robust.